diff --git a/uefi-sct/SctPkg/CommonGenFramework.sh b/uefi-sct/SctPkg/CommonGenFramework.sh
index fb9043b5..1a9f6490 100755
--- a/uefi-sct/SctPkg/CommonGenFramework.sh
+++ b/uefi-sct/SctPkg/CommonGenFramework.sh
@@ -273,6 +273,187 @@ then
     CopyDependency ConfigKeywordHandler
 fi
 
+# *********************************************
+# For BRS SCT
+# *********************************************
+
+if [ $1 = "brs_sct" ]
+then
+    echo "Generating BRS SCT binary"
+    # *********************************************
+    # Copy the UEFI 2.1 Test Cases
+    # *********************************************
+
+    mkdir -p $Framework/SCRT
+    cp $ProcessorType/SCRTDRIVER.efi                           $Framework/SCRT  > NUL
+    cp $ProcessorType/SCRTAPP.efi                              $Framework/SCRT  > NUL
+    cp ../../../SctPkg/Config/Data/SCRT.conf                   $Framework/SCRT  > NUL
+
+    cp $ProcessorType/EfiCompliantBBTest.efi                   $Framework/Test/ > NUL
+
+  # Only include ExeModeBBTest.efi if the file exists
+  if [ -e $ProcessorType/ExeModeBBTest.efi ]; then
+    cp $ProcessorType/ExeModeBBTest.efi                      $Framework/Test/ > NUL
+  fi
+
+    cp $ProcessorType/EventTimerTaskPriorityServicesBBTest.efi $Framework/Test/ > NUL
+    cp $ProcessorType/MemoryAllocationServicesBBTest.efi       $Framework/Test/ > NUL
+    cp $ProcessorType/ProtocolHandlerServicesBBTest.efi        $Framework/Test/ > NUL
+    cp $ProcessorType/ImageServicesBBTest.efi                  $Framework/Test/ > NUL
+    cp $ProcessorType/MiscBootServicesBBTest.efi               $Framework/Test/ > NUL
+
+    cp $ProcessorType/VariableServicesBBTest.efi               $Framework/Test/ > NUL
+    cp $ProcessorType/TimeServicesBBTest.efi                   $Framework/Test/ > NUL
+    cp $ProcessorType/MiscRuntimeServicesBBTest.efi            $Framework/Test/ > NUL
+
+    cp $ProcessorType/BisBBTest.efi                            $Framework/Test/ > NUL
+    cp $ProcessorType/BlockIoBBTest.efi                        $Framework/Test/ > NUL
+    cp $ProcessorType/BlockIo2BBTest.efi                       $Framework/Test/ > NUL
+    cp $ProcessorType/BusSpecificDriverOverrideBBTest.efi      $Framework/Test/ > NUL
+#   cp $ProcessorType/ComponentNameBBTest.efi                  $Framework/Test/ > NUL
+    cp $ProcessorType/DebugPortBBTest.efi                      $Framework/Test/ > NUL
+    cp $ProcessorType/DebugSupportBBTest.efi                   $Framework/Test/ > NUL
+    cp $ProcessorType/DecompressBBTest.efi                     $Framework/Test/ > NUL
+#Note: Device IO BB is deprecated in SCT 2.3
+#   cp $ProcessorType/DeviceIoBBTest.efi                       $Framework/Test/ > NUL
+    cp $ProcessorType/DevicePathBBTest.efi                     $Framework/Test/ > NUL
+    cp $ProcessorType/DevicePathUtilitiesBBTest.efi            $Framework/Test/ > NUL
+    cp $ProcessorType/DevicePathToTextBBTest.efi               $Framework/Test/ > NUL
+    cp $ProcessorType/DevicePathFromTextBBTest.efi             $Framework/Test/ > NUL
+    cp $ProcessorType/DiskIoBBTest.efi                         $Framework/Test/ > NUL
+#   cp $ProcessorType/DriverDiagnosticsBBTest.efi              $Framework/Test/ > NUL
+#   cp $ProcessorType/DriverConfigurationBBTest.efi            $Framework/Test/ > NUL
+    cp $ProcessorType/EbcBBTest.efi                            $Framework/Test/ > NUL
+    cp $ProcessorType/LoadedImageBBTest.efi                    $Framework/Test/ > NUL
+    cp $ProcessorType/LoadFileBBTest.efi                       $Framework/Test/ > NUL
+    cp $ProcessorType/PciIoBBTest.efi                          $Framework/Test/ > NUL
+    cp $ProcessorType/PciRootBridgeIoBBTest.efi                $Framework/Test/ > NUL
+    cp $ProcessorType/PlatformDriverOverrideBBTest.efi         $Framework/Test/ > NUL
+    cp $ProcessorType/PxeBaseCodeBBTest.efi                    $Framework/Test/ > NUL
+#Note: SCSI Passthru Protocol BB is deprecated in SCT 2.3
+#   cp $ProcessorType/ScsiPassThruBBTest.efi                   $Framework/Test/ > NUL
+    cp $ProcessorType/ScsiIoBBTest.efi                         $Framework/Test/ > NUL
+    cp $ProcessorType/ExtScsiPassThruBBTest.efi                $Framework/Test/ > NUL
+    cp $ProcessorType/AtaPassThruBBTest.efi                    $Framework/Test/ > NUL
+    cp $ProcessorType/iScsiInitiatorNameBBTest.efi             $Framework/Test/ > NUL
+    cp $ProcessorType/SerialIoBBTest.efi                       $Framework/Test/ > NUL
+    cp $ProcessorType/SimpleFileSystemBBTest.efi               $Framework/Test/ > NUL
+    cp $ProcessorType/SimpleNetworkBBTest.efi                  $Framework/Test/ > NUL
+    cp $ProcessorType/SimplePointerBBTest.efi                  $Framework/Test/ > NUL
+    cp $ProcessorType/SimpleTextInBBTest.efi                   $Framework/Test/ > NUL
+    cp $ProcessorType/SimpleTextOutBBTest.efi                  $Framework/Test/ > NUL
+#Note: UGA I/O + UGA Draw Protocol BB is deprecated in SCT 2.3
+#   cp $ProcessorType/UgaDrawBbTest.efi                        $Framework/Test/ > NUL
+    cp $ProcessorType/GraphicsOutputBBTest.efi                 $Framework/Test/ > NUL
+    cp $ProcessorType/UnicodeCollation2BBTest.efi              $Framework/Test/ > NUL
+#Note: USB Host Controller Protocol BB is deprecated in SCT 2.3
+#   cp $ProcessorType/UsbHcTest.efi                            $Framework/Test/ > NUL
+    cp $ProcessorType/UsbIoTest.efi                            $Framework/Test/ > NUL
+    cp $ProcessorType/Usb2HcTest.efi                           $Framework/Test/ > NUL
+#    cp $ProcessorType/AuthenticationBBTest.efi                 $Framework/Test/ > NUL
+#   cp $ProcessorType/HashBBTest.efi                           $Framework/Test/ > NUL
+    cp $ProcessorType/TapeBBTest.efi                           $Framework/Test/ > NUL
+    cp $ProcessorType/AcpiTableProtocolBBTest.efi              $Framework/Test/ > NUL
+    cp $ProcessorType/SimpleTextInputExBBTest.efi              $Framework/Test/ > NUL
+    cp $ProcessorType/ComponentName2BBTest.efi                 $Framework/Test/ > NUL
+    cp $ProcessorType/DriverDiagnostics2BBTest.efi             $Framework/Test/ > NUL
+
+    cp $ProcessorType/HIIDatabaseBBTest.efi                    $Framework/Test/ > NUL
+    cp $ProcessorType/HIIStringBBTest.efi                      $Framework/Test/ > NUL
+    cp $ProcessorType/HIIFontBBTest.efi                        $Framework/Test/ > NUL
+    cp $ProcessorType/HIIFontExBBTest.efi                      $Framework/Test/ > NUL
+    cp $ProcessorType/HIIImageBBTest.efi                       $Framework/Test/ > NUL
+    cp $ProcessorType/HIIImageExBBTest.efi                     $Framework/Test/ > NUL
+
+    cp $ProcessorType/AbsolutePointerBBTest.efi                $Framework/Test/ > NUL
+#   cp $ProcessorType/DriverSupportedEfiVersionBBTest.efi      $Framework/Test/ > NUL
+    cp $ProcessorType/PlatformToDriverConfigurationBBTest.efi  $Framework/Test/ > NUL
+    cp $ProcessorType/HIIConfigAccessBBTest.efi                $Framework/Test/ > NUL
+    cp $ProcessorType/HIIConfigRoutingBBTest.efi               $Framework/Test/ > NUL
+    cp $ProcessorType/VlanConfigBBTest.efi                     $Framework/Test/ > NUL
+    cp $ProcessorType/IPsecConfigBBTest.efi                    $Framework/Test/ > NUL
+    cp $ProcessorType/IPsec2BBTest.efi                         $Framework/Test/ > NUL
+    cp $ProcessorType/StorageSecurityCommandBBTest.efi         $Framework/Test/ > NUL
+
+    cp $ProcessorType/FirmwareManagementBBTest.efi             $Framework/Test/ > NUL
+
+    cp $ProcessorType/AdapterInfoBBTest.efi                    $Framework/Test/ > NUL
+    cp $ProcessorType/DiskIo2BBTest.efi                        $Framework/Test/ > NUL
+    cp $ProcessorType/TimeStampBBTest.efi                      $Framework/Test/ > NUL
+    cp $ProcessorType/RandomNumberBBTest.efi                   $Framework/Test/ > NUL
+    cp $ProcessorType/Hash2BBTest.efi                          $Framework/Test/ > NUL
+    cp $ProcessorType/Pkcs7BBTest.efi                          $Framework/Test/ > NUL
+    cp $ProcessorType/ConfigKeywordHandlerBBTest.efi           $Framework/Test/ > NUL
+    cp $ProcessorType/RegularExpressionBBTest.efi              $Framework/Test/ > NUL
+
+    cp $ProcessorType/RamDiskProtocolBBTest.efi                $Framework/Test/ > NUL
+    cp $ProcessorType/NVMEPassThruBBTest.efi                   $Framework/Test/ > NUL
+    cp $ProcessorType/EraseBlockBBTest.efi                     $Framework/Test/ > NUL
+    cp $ProcessorType/SDMMCPassThruBBTest.efi                  $Framework/Test/ > NUL
+    cp $ProcessorType/PartitionInfoBBTest.efi                  $Framework/Test/ > NUL
+    cp $ProcessorType/UFSDeviceConfigBBTest.efi                $Framework/Test/ > NUL
+    cp $ProcessorType/ResetNotificationBBTest.efi              $Framework/Test/ > NUL
+    # Only include RiscVBootProtocolBBTest.efi if the file exists (true on RISCV64)
+    if [ -e $ProcessorType/RiscVBootProtocolBBTest.efi ]; then
+      cp $ProcessorType/RiscVBootProtocolBBTest.efi              $Framework/Test/ > NUL
+    fi
+
+    # *********************************************
+    # BRS test cases
+    # *********************************************
+
+    # Check if EFI Specification Version
+    cp $ProcessorType/EfiSpecVerLvlBBTest.efi                  $Framework/Test/ > NUL
+    cp $ProcessorType/BrsBootServicesBBTest.efi                $Framework/Test/ > NUL
+    cp $ProcessorType/RequiredUefiProtocolsBBTest.efi          $Framework/Test/ > NUL
+    cp $ProcessorType/BRSIRuntimeServicesBBTest.efi            $Framework/Test/ > NUL
+
+    # *********************************************
+    # Copy ENTS binary
+    # *********************************************
+
+    cp $ProcessorType/SerialMonitor.efi                $Framework/Ents/Support/ > NUL
+    cp $ProcessorType/ManagedNetworkMonitor.efi        $Framework/Ents/Support/ > NUL
+    cp $ProcessorType/IP4NetworkMonitor.efi            $Framework/Ents/Support/ > NUL
+    cp $ProcessorType/Eftp.efi                         $Framework/Ents/Support/ > NUL
+
+    cp $ProcessorType/BootService_ENTSTest.efi         $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/RuntimeService_ENTSTest.efi      $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/GenericService_ENTSTest.efi      $Framework/Ents/Test/    > NUL
+
+    cp $ProcessorType/SimpleNetwork_ENTSTest.efi       $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/PXEBaseCode_ENTSTest.efi         $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/Mnp*_ENTSTest.efi                $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/Arp*_ENTSTest.efi                $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/Ip4*_ENTSTest.efi                $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/Ip6*_ENTSTest.efi                $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/Udp4*_ENTSTest.efi               $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/Udp6*_ENTSTest.efi               $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/Dhcp4*_ENTSTest.efi              $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/Dhcp6*_ENTSTest.efi              $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/Mtftp4*_ENTSTest.efi             $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/Mtftp6*_ENTSTest.efi             $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/Tcp4*_ENTSTest.efi               $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/Tcp6*_ENTSTest.efi               $Framework/Ents/Test/    > NUL
+    cp $ProcessorType/Http*_ENTSTest.efi               $Framework/Ents/Test/    > NUL
+
+    # *********************************************
+    # Copy the test dependency files
+    # *********************************************
+
+    CopyDependency EfiCompliant
+    CopyDependency ProtocolHandlerServices
+    CopyDependency ImageServices
+    CopyDependency Decompress
+    CopyDependency DeviceIo
+    CopyDependency Ebc
+    CopyDependency LoadedImage
+    CopyDependency PciIo
+    CopyDependency PciRootBridgeIo
+    CopyDependency PxeBaseCode
+    CopyDependency ConfigKeywordHandler
+fi
+
 # *********************************************
 # For IHV SCT
 # *********************************************
diff --git a/uefi-sct/SctPkg/Config/Data/Category.ini b/uefi-sct/SctPkg/Config/Data/Category.ini
index b34d35db..39a9bf8b 100644
--- a/uefi-sct/SctPkg/Config/Data/Category.ini
+++ b/uefi-sct/SctPkg/Config/Data/Category.ini
@@ -88,6 +88,20 @@ InterfaceGuid = 71652D04-BF38-434A-BCB8-6547D7FD8384
 Name          = GenericTest\FloatingPointABITest
 Description   =
 
+[Category Data]
+Revision      = 0x00010000
+CategoryGuid  = d6e40d12-7a51-4454-b4fd-4c93887ed6cf
+InterfaceGuid = 71652D04-BF38-434A-BCB8-6547D7FD8384
+Name          = GenericTest\TestEfiSpecVerLvl
+Description   =
+
+[Category Data]
+Revision      = 0x00010000
+CategoryGuid  = 6b4d5d18-f55f-4fc5-8496-f10022375c04
+InterfaceGuid = 71652D04-BF38-434A-BCB8-6547D7FD8384
+Name          = GenericTest\BrsiRequiredUefiProtocols
+Description   = Checks that UEFI protocols required by BRS are present.
+
 [Category Data]
 Revision      = 0x00010000
 CategoryGuid  = E9EF7553-F833-4E56-96E8-38AE679523CC
@@ -130,6 +144,13 @@ InterfaceGuid = E9EF7553-F833-4E56-96E8-38AE679523CC
 Name          = BootServicesTest\MiscBootServicesTest
 Description   =
 
+[Category Data]
+Revision      = 0x00010000
+CategoryGuid  = 236da812-2002-4ad9-884d-058fd2dd1386
+InterfaceGuid = E9EF7553-F833-4E56-96E8-38AE679523CC
+Name          = BootServicesTest\BrsBootServices
+Description   = Tests BRS boot services requirements.
+
 [Category Data]
 Revision      = 0x00010000
 CategoryGuid  = AFF115FB-387B-4C18-8C41-6AFC7F03BB90
@@ -164,6 +185,13 @@ CategoryGuid  = 65AB91D6-AB78-4A5D-BA80-1ABD2358C3A4
 InterfaceGuid = AFF115FB-387B-4C18-8C41-6AFC7F03BB90
 Name          = RuntimeServicesTest\MiscRuntimeServicesTest
 Description   =
+ 
+[Category Data]
+Revision      = 0x00010000
+CategoryGuid  = 1C3C483B-3BA3-42AB-80EC-5AE79D31F193
+InterfaceGuid = AFF115FB-387B-4c18-8C41-6AFC7F03BB90
+Name          = RuntimeServicesTest\BRSIRuntimeServicesTest
+Description   = RRSI Runtime Services Tests
 
 [Category Data]
 Revision      = 0x00010000
@@ -1026,9 +1054,9 @@ InterfaceGuid = 8D59D32B-C655-4AE9-9B15-F25904992A43
 Name          = IHV\ConsoleSupportTest\AbsolutePointerProtocolTest
 Description   = Absolute Pointer Protocol Test on IHV Drivers
 
-[Category Data]
-Revision      = 0x00010000
-CategoryGuid  = CCD15FEC-6F73-4EEC-8395-3E69E4B940BF
-InterfaceGuid = CCD15FEC-6F73-4EEC-8395-3E69E4B940BF
-Name          = RiscvBootProtocolTest
-Description   = RISC-V Boot Protocol Test
+[Category Data]
+Revision      = 0x00010000
+CategoryGuid  = CCD15FEC-6F73-4EEC-8395-3E69E4B940BF
+InterfaceGuid = CCD15FEC-6F73-4EEC-8395-3E69E4B940BF
+Name          = RiscvBootProtocolTest
+Description   = RISC-V Boot Protocol Test
