From 4a35758c039b083f0f3ad1d6338390e50dc6040c Mon Sep 17 00:00:00 2001
From: Yin Wang <yin.wang@intel.com>
Date: Wed, 12 Apr 2023 11:17:02 +0800
Subject: [PATCH] update gnulib to gitbub to fix network issue

---
 bootstrap.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bootstrap.conf b/bootstrap.conf
index 6b043fc35..44104dd0a 100644
--- a/bootstrap.conf
+++ b/bootstrap.conf
@@ -15,7 +15,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-
+GNULIB_URL=https://github.com/coreutils/gnulib.git
 GNULIB_REVISION=d271f868a8df9bbec29049d01e056481b7a1a263
 
 # gnulib modules used by this package.
-- 
2.34.1

