From fa4ae0c1c2048154b141ddf35fe45c0b520a7650 Mon Sep 17 00:00:00 2001
From: Haibo Xu <haibo1.xu@intel.com>
Date: Wed, 19 Jul 2023 14:53:28 +0800
Subject: [PATCH 2/6] acpi: rsdp: Add xsdt_address check for riscv

---
 src/acpi/rsdp/rsdp.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/acpi/rsdp/rsdp.c b/src/acpi/rsdp/rsdp.c
index 419016f3..72705b36 100644
--- a/src/acpi/rsdp/rsdp.c
+++ b/src/acpi/rsdp/rsdp.c
@@ -165,6 +165,11 @@ static int rsdp_test1(fwts_framework *fw)
 			passed = true;
 		break;
 
+	case FWTS_ARCH_RISCV:
+		if (rsdp->xsdt_address != 0)
+			passed = true;
+		break;
+
 	default:
 		passed = true;
 		fwts_log_advice(fw,
-- 
2.34.1

