From ef035c2fa9bff872552ba697c57a8c7a95675c53 Mon Sep 17 00:00:00 2001
From: Joseph Hemann <Joseph.hemann@arm.com>
Date: Thu, 19 Aug 2021 19:42:44 -0500
Subject: [PATCH 36/37] uefi-sct/SctPkg: Remove warnings in TCG2 protocol test

Signed-off-by: Joseph Hemann <Joseph.hemann@arm.com>
Signed-off-by: Stuart Yoder <stuart.yoder@arm.com>
---
 .../BlackBoxTest/TCG2ProtocolBBTestConformance.c            | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/TCG2Protocol/BlackBoxTest/TCG2ProtocolBBTestConformance.c b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/TCG2Protocol/BlackBoxTest/TCG2ProtocolBBTestConformance.c
index 31b31f87..7fabce09 100644
--- a/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/TCG2Protocol/BlackBoxTest/TCG2ProtocolBBTestConformance.c
+++ b/uefi-sct/SctPkg/TestCase/UEFI/EFI/Protocol/TCG2Protocol/BlackBoxTest/TCG2ProtocolBBTestConformance.c
@@ -882,7 +882,7 @@ BBTestHashLogExtendEventConformanceTestCheckpoint4 (
     AssertionType = EFI_TEST_ASSERTION_FAILED;
   }
 
-  Status = SctStrCmp(EventLogHeaderSpecEvent->signature, data);
+  Status = SctAsciiStrCmp(EventLogHeaderSpecEvent->signature, data);
   // Verify EventLog Signature
   if (Status != EFI_SUCCESS) {
     StandardLib->RecordMessage (
@@ -966,9 +966,9 @@ BBTestSubmitCommandConformanceTestCheckpoint1 (
   Status = TCG2->SubmitCommand (
                            TCG2,
                            sizeof(GET_RANDOM_COMMAND),
-                           &CommandInput,
+                           (UINT8 *)&CommandInput,
                            sizeof(GET_RANDOM_RESPONSE),
-                           &CommandResponse);
+                           (UINT8 *)&CommandResponse);
 
   AssertionType = EFI_TEST_ASSERTION_PASSED;
 
-- 
2.17.1

